code/modules/materials/definitions/materials_ice.dm

```diff
@@ -0,0 +1,88 @@
+/decl/material/aspium_ice
+	display_name = "aspium ice"
+	stack_type = null
+	icon_colour = "#a5f2f3"
+	chemical_makeup = list(
+		/decl/reagent/fuel/hydrazine 	= 0.3,
+		/decl/reagent/water				= 0.7
+	)
+	ore_spread_chance = 25
+	ore_scan_icon = "mineral_common"
+	ore_name = "aspium ice"
+	ore_icon_overlay = "lump"
+	removed_by_welder = TRUE
+	value = 0.4
+	sparse_material_weight = 2
+	rich_material_weight = 37
+
+/decl/material/lukrite_ice
+	display_name = "lukrite ice"
+	stack_type = null
+	icon_colour = "#a5f2f3"
+	chemical_makeup = list(
+		/decl/reagent/sulfur		 	= 0.4,
+		/decl/reagent/water				= 0.2,
+		/decl/reagent/acid				= 0.4
+	)
+	ore_spread_chance = 25
+	ore_scan_icon = "mineral_common"
+	ore_name = "lukrite ice"
+	ore_icon_overlay = "lump"
+	removed_by_welder = TRUE
+	value = 0.3
+	sparse_material_weight = 27
+	rich_material_weight = 22
+
+/decl/material/rubenium_ice
+	display_name = "rubenium ice"
+	stack_type = null
+	icon_colour = "#a5f2f3"
+	chemical_makeup = list(
+		/decl/reagent/radium		 	= 0.4,
+		/decl/reagent/water				= 0.4,
+		/decl/reagent/acetone			= 0.2
+	)
+	ore_spread_chance = 25
+	ore_scan_icon = "mineral_common"
+	ore_name = "rubenium ice"
+	ore_icon_overlay = "lump"
+	removed_by_welder = TRUE
+	value = 0.4
+	sparse_material_weight = 35
+	rich_material_weight = 12
+
+/decl/material/trigarite_ice
+	display_name = "trigarite ice"
+	stack_type = null
+	icon_colour = "#a5f2f3"
+	chemical_makeup = list(
+		/decl/reagent/acid/hydrochloric	= 0.2,
+		/decl/reagent/water				= 0.2,
+		/decl/reagent/mercury			= 0.6
+	)
+	ore_spread_chance = 25
+	ore_scan_icon = "mineral_common"
+	ore_name = "trigarite ice"
+	ore_icon_overlay = "lump"
+	removed_by_welder = TRUE
+	value = 0.5
+	sparse_material_weight = 38
+	rich_material_weight = 23
+
+/decl/material/ediroite_ice
+	display_name = "ediroite ice"
```

Comment:
If you made a `/decl/material/ice` and moved all the shared vars onto it you could make this file a lot smaller.